docs/bls-keys/dd.md

```diff
@@ -0,0 +1,103 @@
+- `GSE.getAttesterFromIndexAtTime` -> `GSE.getAttesterBLSPubKeyFromIndexAtTime`
+
+## Security considerations
+
+When registering a new key, we first need to validate that it is a correct public key in the curve. Then, as described in [this article](https://www.zellic.io/blog/bls-signature-versatility/#the-pitfall-of-multi-signatures), we need to guard against rogue key attacks. We have two options for that:
+
+1. Require a proof of possession, validated along with the key during registration
+2. Use a modified aggregate public key, which should only require 2 scalar multiplications
+
+While the 2nd approach has not been implemented according to the article linked above, it sounds simple enough and the additional gas cost is low.
+
+## Open questions
+
+### Is the choice of curve correct?
+
+We defaulted to the curve used by ZCash and Ethereum consensus, but there may be other curves that are a better fit.
+
+### When should we validate the public key is valid?
+
+The innermost contract (GSE) seems to be the most secure place where to execute this validation.
+
+On the other hand, if we know that all entries to GSE have to go through the staking queue, we could instead validate early in the `enqueue`, which would also allow us to change curves in the future without having to modify the GSE.
+
+To favor failing fast, I prefer validating in the queue, assuming we are certain that the queue is always present.
+
+### How to prevent against the rogue key attack?
+
+As discussed above in the Security Considerations section, we need to decide between requiring a proof of possession or using a modified aggregate public key.
+
+### How do we generate the BLS private key for validators?
+
+We could derive the BLS private key from their ECDSA private key, potentially from having them sign a given message so we do not need direct access to the ECDSA private key. This means that validators do not need to store two separate private keys. Note that this requires **deterministic** ECDSA signatures, which not every hardware wallet supports.
+
+We need to validate whether this is feasible and secure.
+
+### Should we support rotation of the BLS keys?
+
+If we do not default to generating the BLS private key from the validator ECDSA private key, should we allow validators to change them? It's unclear if rotating should involve a forced delay, so that a validator cannot change their BLS key halfway through an epoch.
+
+To avoid additional complexity, I suggest not to. Worst case, a validator can exit and re-enter with different BLS keys.
+
+### Should we support changing key family in the future?
+
+If we find a better signing algorithm or curve in the future, we have no easy way for validators to switch key scheme. This would require each validator to have both set of keys (the old and new ones) registered simultaneously, so when the canonical rollup instance is changed, the new one can immediately access the new keys.
```

Comment:
I would say DONT SUPPORT THIS. This is something where I would say it is a new GSE that is needed. If the structure can be different between which rollup you registered at, any of the "automatic moving" seems like it could go straight to hell if a new version is expecting a different type because people that are automatically moved might not follow similar constraints to the format or data. 

Essentially, a consumer of the values in the GSE should be able to assume that the data in it follow some specific format and have passed certain checks.